python/py-numpy/Portfile

```diff
@@ -286,38 +72,89 @@ variant."
             reinplace   "s|---|\\\\.f|" ${worksrcpath}/f-wrapper
             reinplace   "s|&&&|${prefix}|" ${worksrcpath}/f-wrapper
         }
+    }
 
-        if {[variant_isset atlas]} {
-            # We must link against libSatlas or libTatlas, not libAtlas
-            if {[file exists ${prefix}/lib/libtatlas.dylib]} {
-                reinplace -E \
-                    "s|_lib_atlas = \\\['atlas'\\\]|_lib_atlas = \\\['tatlas'\\\]|" \
-                    ${worksrcpath}/numpy/distutils/system_info.py
-            } elseif {[file exists ${prefix}/lib/libsatlas.dylib]} {
-                reinplace -E \
-                    "s|_lib_atlas = \\\['atlas'\\\]|_lib_atlas = \\\['satlas'\\\]|" \
-                    ${worksrcpath}/numpy/distutils/system_info.py
-            } else {
-                return -code error "Unable to find Atlas dylibs. Bailing out."
-            }
+    depends_build-append \
+                        path:bin/pkg-config:pkgconfig
+
+    build.env-append    CYTHON=${prefix}/bin/cython-${python.branch}
+
+    # do not allow fallback to (slow) NumPy-internal routines if no external BLAS/LAPACK libraries are found
+    build.args-append -Csetup-args=-Dallow-noblas=false
+
+    compilers.setup     -clang -gcc44 -gcc45 \
+                        -gcc46 -gcc47 -gcc48 -g95
+
+    # https://trac.macports.org/ticket/67136
+    depends_run-append  port:py${python.version}-oldest-supported-numpy
+
+    depends_lib-append  port:fftw-3 \
+                        path:bin/cython-${python.branch}:py${python.version}-cython
+
+    variant accelerate conflicts openblas flexiblas blis description "Use Apples Accelerate Framework" {
+        build.args-append -Csetup-args=-Dblas=accelerate -Csetup-args=-Dlapack=accelerate
+    }
+
+    variant openblas conflicts accelerate flexiblas blis description "Use MacPorts OpenBLAS Libraries" {
+        depends_lib-append  path:lib/libopenblas.dylib:OpenBLAS
+
+        # see: https://trac.macports.org/ticket/66685
+        supported_archs     arm64 ppc ppc64 x86_64
```

Comment:
Is OpenBLAS broken on i386? Perhaps raise to upstream, if not yet. They are very helpful.

Reply:
I have no idea, that Trac ticket isn't very clear and it seems this change was mainly "I don't know but if it doesn't work let's disable it". I might just remove that and see how it goes (I can personally not test).

Reply:
I am generally in favor of avoiding outright blacklists unless we know for sure something is broken beyond repair. After all, it can be useful to get a bug report from i386, rather than prohibiting to try the build.
Or maybe display a message to a user that it is disabled temporarily.

Reply:
just to be clear: this was not my addition, it was already in the original Portfile.